src/shipping/strategies/amazon/amazon-pay-shipping-strategy.spec.ts

```diff
@@ -269,4 +269,23 @@ describe('AmazonPayShippingStrategy', () => {
         expect(store.dispatch).toHaveBeenCalledWith(action);
         expect(output).toEqual(store.getState());
     });
+
+    it('updates address with provided custom fields and existing address', async () => {
+        const strategy = new AmazonPayShippingStrategy(store, consignmentActionCreator, paymentMethodActionCreator, remoteCheckoutActionCreator, scriptLoader);
+        const options = {};
+        const address = getShippingAddressWithCustomFields();
+        const action = of(createAction(ConsignmentActionType.UpdateConsignmentRequested));
+
+        jest.spyOn(consignmentActionCreator, 'updateAddress')
+            .mockReturnValue(action);
+
+        jest.spyOn(store, 'dispatch');
+
+        const output = await strategy.updateAddress(address, options);
+
+        expect(store.getState().shippingAddress.getShippingAddress()).not.toEqual(address);
```

Comment:
what I was suggesting is to mock `store.getState().shippingAddress.getShippingAddress()` to return addressA
and call `strategy.updateAddress(address)` with addressB.

Then
```
 expect(consignmentActionCreator.updateAddress).toHaveBeenCalledWith({
 ...addressA,
  customFields: addressB.customFields,
}, options);
```

Reply:
Sure